docs/src/main/sphinx/functions/datetime.md

```diff
@@ -571,12 +571,26 @@ This is an alias for {func}`year_of_week`.
 :::
 
 :::{function} timezone(timestamp(p) with time zone) -> varchar 
-Returns the IANA timezone ID from `timestamp(p) with time zone`. 
+
+Returns the timezone identifier from `timestamp(p) with time zone`. The format
+of the returned identifier is identical to the [format used in the input
+timestamp](timestamp-p-with-time-zone-data-type):
+
+```sql
+SELECT timezone(TIMESTAMP '2024-01-01 12:00:00 Asia/Tokyo'); -- Asia/Tokyo
+SELECT timezone(TIMESTAMP '2024-01-01 12:00:00 +01:00'); -- +01:00
+SELECT timezone(TIMESTAMP '2024-02-29 12:00:00 UTC'); -- UTC
+```
 :::
 
 :::{function} timezone(time(p) with time zone) -> varchar
 :no-index:
-Returns the IANA timezone ID from `time(p) with time zone`. 
+Returns the timezone identifier from a `time(p) with time zone` as numeric UTC
+offset value.
```

Comment:
Can you confirm that the docs are correct here and in general @martint .. the time zone with time only supports offset value .. I vaguely remember discussing this in the past and I think its correct.. although a bit weird.

Reply:
Yes, time w/tz only supports offsets because political time zones depend on the specific date they are associated with.

Reply:
You could use the same phrasing as above: "identical to the format used in the input time", since they can only be constructed with offsets.

Reply:
yeah .. but that doesnt call the weird difference out.

Reply:
Also .. should we add that info about political time zones to the docs? I could add that as separate commit to this PR

Reply:
I don't see that as a weird difference. Timestamps w/ timezone and time w/ timezone are different types with different constraints.

Reply:
Done